src/ws/cockpitauth.c

```diff
@@ -1047,6 +1047,13 @@ cockpit_session_launch (CockpitAuth *self,
       return NULL;
     }
 
+  /* this might be unset, which means "allow if cockpit-ssh is installed"; if it isn't, this will fail later on */
+  if (host && !cockpit_conf_bool ("WebService", "LoginTo", TRUE)) {
```

Comment:
This looks sensible.

There are four cases where we could end up using cockpit-ssh.  Two of those involve explicit config and one of them involves a commandline option.  This should successfully neutralize the only case the user controls.

Reply:
So that's (1) login page with "Connect to:" (potentially dangerous), (2) shell's remote host (dangerous in a different way due to non-isolated frames, but not internal network scanning), (3) `cockpit-ws --local-session` (harmless). What is (4)?